Plugins/NativeCamera/Source/Android/NativeCameraImpl.cpp

```diff
@@ -217,56 +178,42 @@ namespace Babylon::Plugins
     };
 
     Camera::Impl::Impl(Napi::Env env, bool overrideCameraTexture)
-        : m_deviceContext{nullptr}
-        , m_env{env}
-        , m_overrideCameraTexture{overrideCameraTexture}
+        : m_implData{std::make_unique<ImplData>(env, overrideCameraTexture)}
     {
         if (API_LEVEL < 24 && !overrideCameraTexture)
         {
             throw std::runtime_error{"Android Platform level < 24. Only camera texture override is available."};
         }
+
+        if (!overrideCameraTexture)
+        {
+            m_implData->cameraManager = GET_CAMERA_FUNCTION(ACameraManager_create)();
```

Comment:
Any reason to do this here instead of in the implData constructor?

Reply:
It needs to be guarded by the API check... but I can do that in the constructor as well. I'll move it up there.